src/hepattn/models/maskformer.py

```diff
@@ -61,10 +61,37 @@ def __init__(
         self.input_sort_field = input_sort_field
         self.use_attn_masks = use_attn_masks
         self.use_query_masks = use_query_masks
+        self.log_attn_mask = log_attn_mask
+        self.step_ = 0
+
+    def get_last_attention_mask(self):
+        """Get the last attention mask that was stored for logging.
+        Returns
+        -------
+        tuple or None
+            A tuple of (attention_mask, step, layer) if available, None otherwise.
+        """
+        if hasattr(self, '_last_attn_mask'):
+            return (
+                self._last_attn_mask,
+                getattr(self, '_last_attn_mask_step', 0),
+                getattr(self, '_last_attn_mask_layer', 0)
+            )
+        return None
+
+    def clear_last_attention_mask(self):
+        """Clear the stored attention mask after logging."""
+        if hasattr(self, '_last_attn_mask'):
+            del self._last_attn_mask
+        if hasattr(self, '_last_attn_mask_step'):
+            del self._last_attn_mask_step
+        if hasattr(self, '_last_attn_mask_layer'):
+            del self._last_attn_mask_layer
```

Comment:
I would remove the boilerplate here and just directly access/clear the member from the callback.